source/geometries/BlackBox.cc

```diff
@@ -0,0 +1,251 @@
+// ----------------------------------------------------------------------------
+// nexus | BlackBox.cc
+//
+// Sapphire disk and DICE board with configurable mask thickness parallelly
+// placed in a black box.
+//
+// The NEXT Collaboration
+// ----------------------------------------------------------------------------
+
+#include "BlackBox.h"
+#include "BlackBoxSiPMBoard.h"
+#include "FactoryBase.h"
+
+#include "MaterialsList.h"
+#include "IonizationSD.h"
+#include "OpticalMaterialProperties.h"
+#include "Visibilities.h"
+
+#include <G4Box.hh>
+#include <G4Tubs.hh>
+#include <G4GenericMessenger.hh>
+#include <G4Orb.hh>
+#include <G4NistManager.hh>
+#include <G4LogicalVolume.hh>
+#include <G4PVPlacement.hh>
+#include <G4Material.hh>
+#include <G4ThreeVector.hh>
+#include <G4VisAttributes.hh>
+#include <G4Colour.hh>
+#include <G4SDManager.hh>
+#include <G4UserLimits.hh>
+#include <Randomize.hh>
+#include <G4OpticalSurface.hh>
+#include <G4LogicalSkinSurface.hh>
+#include <G4RotationMatrix.hh>
+#include <G4SystemOfUnits.hh>
+
+namespace nexus {
+
+  REGISTER_CLASS(BlackBox, GeometryBase)
+
+  BlackBox::BlackBox():
+    GeometryBase(),
+    world_z_ (3. * m),
+    world_xy_ (2. *m),
+    box_z_ (2. * m),
+    box_xy_ (1. *m),
+    specific_vertex_(0,0,0),
+    dice_board_x_pos_(0),
+    dice_board_y_pos_(0),
+    dice_board_z_pos_(-49. *cm),
+    rotation_(0),
+    mask_thickn_(),
+    hole_diameter_(1),
+    hole_x_(1),
+    hole_y_(1),
+
+    visibility_(0)
```

Comment:
Move the initialization of the `visibility_` variable up, to follow the same order as the declaration in the header file. This way, the warning that appear at compilation time will disappear.